shared/django_apps/core/models.py

```diff
@@ -430,11 +430,17 @@ def should_write_to_storage(self) -> bool:
     _flare = models.JSONField(db_column="flare", null=True)
     _flare_storage_path = models.URLField(db_column="flare_storage_path", null=True)
     flare = ArchiveField(
-        should_write_to_storage_fn=should_write_to_storage, default_value_class=dict
+        should_write_to_storage_fn=should_write_to_storage,
+        default_value_class=dict,
     )
 
     def save(self, *args, **kwargs):
         self.updatestamp = timezone.now()
+        if self.state != PullStates.OPEN.value and self.flare:
+            # flare is used to draw graphs, can be quite large, so dump it when it's no longer needed
+            # flare can be None or {} to indicate empty
+            # this gets set in the db as {} since that is the default value for the field
+            self.flare = None
```

Comment:
We should be careful with this one. I believe even accessing to check `self.flare` will fetch that file from storage (if its not being fetched ahead of time already).

And then deleting that file incurs a ~200ms GCS DELETE operation.

In particular these GCS requests are slow and might cause even more problems than a trigger unsetting a field that I believe was never written to anyway since the storage migration.

As you suggested, this might require a cronjob to asynchronously clean up storage.
Another thing that comes to mind here is that we probably never removed these things from GCS ever since that storage migration. So the trigger was completely ineffective for a long time probably.

Reply:
thank you times a million for this insight - I wasn't understanding what was happening with the `ArchiveField` in relation to GCS storage. If this is the case, we should make this a nightly task that runs after hours. Going to close this and reopen later